sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -1100,6 +1100,12 @@ struct _pi_event : _pi_object {
   // actions are performed only once.
   //
   bool CleanedUp = {false};
+
+  // Indicates that this PI event had already completed in the sense
+  // that no other synchromization is needed. Note that the underlying
+  // L0 event (if any) is not guranteed to have been signalled, or
+  // being visible to the host at all.
+  bool Completed = {false};
```

Comment:
NOTE: this is strictly needed to handle PI events that don't have host-visible L0 event (since they are handled on the host), but it is also an optimization to not sync the same PI event multiple times.